File: 1291-sequential-digits/1291-sequential-digits.java
class Solution {
     public List<Integer> sequentialDigits(int low, int high) {
        List<Integer> ans = new ArrayList<>();
        String st = String.valueOf(low);
        String end = String.valueOf(high);
        for (int i = st.length(); i <= end.length(); i++) {
            for (int j = 1; j <= 9; j++) {
                StringBuilder curr = new StringBuilder();
                for (int k = j; k < Math.min(10, j + i); k++) {
                    curr.append(k);
                }
                if (curr.length() != i) {
                    continue;
                }
                int num = Integer.parseInt(curr.toString());
                if (num >= low && num <= high) {
                    ans.add(num);
                }
            }
        }
        ans.sort(null);
        return ans;
    }
}

File: 1291-sequential-digits/1291-sequential-digits.cpp
class Solution {
public:
    vector<int> sequentialDigits(int low, int high) {
        vector<int>ans;
        string st=to_string(low);
        string end=to_string(high);
        for(int i=st.length();i<=end.length();i++){
            for(int j=1;j<=9;j++){
                string curr="";
                for(int k=j;k<min(10,j+i);k++){
                    curr+=to_string(k);
                }
                if(curr.length()!=i)continue;
                int num=stoi(curr);
                if(num>=low && num<=high)ans.push_back(num);
            }
        }
        sort(ans.begin(),ans.end());
        return ans;
    }
};